### ISSUE TITLE
[Feature Request] Add delay option after job fail

### ISSUE THREAD

**Issue by mix2zeta** [Proteus-tech/mod-ngarn#3]
Currently job will do immediately after fail if it next in queue, It should have some delay after some consecutive fail. like 1 hour after 10 fail, 24 hr after 50 fail and it should not stop what ever how many fail.

**Comment by KayEss** [comment]
Yup, you want some exponential backoff, ideally you'd have a max delay. You can find an example in the [django-async](https://github.com/KayEss/django-async/blob/master/async/models.py#L228) code. it doesn't have the max, but ideally there would be one.
